Sources/Widgets/Widgets3D/EllipseWidget/behavior.js

```diff
@@ -9,38 +8,26 @@ export default function widgetBehavior(publicAPI, model) {
 
   model.classHierarchy.push('vtkEllipseWidgetProp');
 
-  publicAPI.setBounds = (bounds) => {
-    if (superClass.setBounds) {
-      superClass.setBounds(bounds);
+  publicAPI.setCorners = (point1, point2) => {
+    if (superClass.setCorners) {
+      superClass.setCorners(point1, point2);
     }
 
-    const center = vtkBoundingBox.getCenter(bounds);
-    const scale3 = vtkBoundingBox.computeScale3(bounds);
-
-    model.shapeHandle.setOrigin(center);
-    model.shapeHandle.setScale3(scale3);
-  };
-
-  publicAPI.setBoundsFromRadius = (center, pointOnCircle) => {
-    const radius = vec3.distance(center, pointOnCircle);
-
-    publicAPI.setBounds([
-      center[0] - radius,
-      center[0] + radius,
-      center[1] - radius,
-      center[1] + radius,
-      center[2] - radius,
-      center[2] + radius,
-    ]);
-  };
-
-  publicAPI.setBoundsFromDiameter = (point1, point2) => {
     const center = [
       0.5 * (point1[0] + point2[0]),
       0.5 * (point1[1] + point2[1]),
       0.5 * (point1[2] + point2[2]),
     ];
 
-    publicAPI.setBoundsFromRadius(center, point1);
+    const d = [0, 0, 0];
+    vec3.subtract(d, point2, center);
+
+    const horizontal = model.shapeHandle.getRight();
+    const vertical = model.shapeHandle.getUp();
+    const sh = vec3.dot(d, horizontal);
+    const sv = vec3.dot(d, vertical);
+
+    model.shapeHandle.setOrigin(center);
+    model.shapeHandle.setScale3([sh, sv, 1]);
```

Comment:
why not support z scale ? (in case corner points are not within the same plane)

Reply:
I'm having issues finishing that part up: we have two corners and they should be on the same plane as the right/up plane that's parallel to the camera plane. The scaling here `[sh, sv, 1]` is actually in that plane coordinate system, but I am not sure if `scale3` should be in that local coordinate system, or in the world XYZ coordinate system...?

Whichever it is/should be, those current values coupled with how the orientation is applied to it in the `CircleContextRepresentation` and in the `Glyph3DMapper` just doesn't quite work yet... an attempt was made in the last `WIP` commit of this PR to fix the orientation in those classes, but my brain was fried last Friday evening and I could not make that work.
cc: @floryst who witnessed my brain-melting 🧠🔥

Reply:
The scale is in local space, not world space, so we can actually put any value in that third component, it will always be ignored because the `circleContextRepresentation` is a 2D representation.

We've investigated more with @floryst and came up with what can be seen in [this commit](https://github.com/Kitware/vtk-js/pull/1422/commits/664e1eac81ccf4b5922054022c001ed937cfa621):
- removed some code from the `circleContextRepresentation` (no need to reorient the scale)
- added `MATRIX` to `OrientationModes` that is implemented in `Glyph3DMapper` and used by `CircleContextRepresentation`

Debug only:
- modified the `Glyph3DMapper` example to observe the scaling behavior and test/compare `OrientationModes.MATRIX` and `OrientationModes.ROTATION` -> they work 🎉
- modified the `ShapeWidget` example to add an orientation widget to try to understand why the `CircleContextRepresentation` is not shaping properly -> still haven't figured it out 🤕

@floryst plans to look at it again at some point to figure that out.